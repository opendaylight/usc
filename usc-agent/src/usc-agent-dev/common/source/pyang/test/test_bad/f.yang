module f {
    namespace "urn:f";
    prefix f;
    
    grouping xxx {
        leaf z4 {
            type int32;
        }
    }

  list gg { // error: needs key
    leaf foo {
      type int32;
    }
  }

  container x {
    config false;
    list gg2 { 
      leaf foo {
        type int32;
      }
    }
    leaf bar {
      config true; // error: config true when parent config is false
      type string;
    }

  }

  list ff {
    typedef ee { // warning: not used
      type empty;
    }

    key "z1 z4 z2"; // error: z2 is empty

    unique "foo"; // error: foo does not exist

    leaf z1 {
      type int32;
    }

    leaf z2 {
      type ee;
    }

    uses xxx;

  }


}
