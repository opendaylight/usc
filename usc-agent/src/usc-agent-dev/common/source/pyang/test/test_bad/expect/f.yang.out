f.yang:11: error: NEED_KEY
f.yang:25: error: INVALID_CONFIG
f.yang:36: error: BAD_TYPE_IN_KEY
f.yang:38: error: BAD_UNIQUE_PART
